docs/3-Files.md

```diff
@@ -0,0 +1,88 @@
+## Files for storing parameters
+
+Currently, two formats are supported for persistent storage of parameters: 
+[`toml`](https://toml.io/en/) and [`json`](https://www.json.org/).
+
+The `toml` format is human-oriented, flexible, standardardized and not overly complex.
+It supports comments, and hence parameters can be easily documented in a toml file.
+However, when parameters are initialized by reading from a toml file, then changed
+(because it concerns settings) and subsequently written to file again for persistance,
+the comments will be lost. Therefore, it does not make much sense to add documenting
+comments to a settings file. Because config parameters are read-only, comments in a
+config file will not get lost and do make sense. Because of this, the default format for
+storing config parameters is `toml`. The name of a config file equals `config.toml` by
+default.
+
+JSON is a standardized, lightweight data-interchange format that is easy for machines to
+parse and generate. It is a bit less straightforward to document parameters in this
+format, which makes it less human-oriented than `toml`, but it is used widely for data
+interchange between automated systems. Because of this, the default format for storing
+settings parameters is `json`. The name of a settings file equals `settings.json` by
+default.
+
+The examples introduced in the previous section can for example be initialized with the
+following files.
+
+=== "`config.toml` file for the configuration example"
+    ```toml
+    [section1]
+    # Use this section to set the config that you prefer by editing the file
+    # field1 has default value 'field1'
+    field1 = "my own version of the first field"
+    # field2 has default value 2
+    field2 = 22
+    ```
+
+=== "`settings.json` file for the settings example"
+    ```json
+    {
+        "basics": {
+            "name": "the stored name",
+            "totals": 3
+        }
+    }
+    ```
+
+
+The fieldnames of a Container class are found as:
+
+- the section names in the `toml` file
+- the member names of the root object in the `json` file
+
+The fieldnames of a Section class are found as:
+
+- the fieldnames inside the corresponding section in the `toml` file
+- the member names of the object that is the value of the name that represents the
+  section in the `json` file
+
+The order of sections and/or fields in the file does not have to adhere to the order
+in which fields have been specified in the Container - or Section classes.
+
+Presence of sections and/or fields in the files that are not defined in the classes goes
+by silently.
+
+Fields or complete sections defined in the classes can be absent in the files as long as
+default values have been specified for all fields that have been left out.
```

Comment:
maybe add: `, otherwise a ValueError is raised`

Reply:
Consequently something I'm now thinking of. One annoyance I can have with libraries, is raising a single error every time.

Do we actually collect all them missing / wrong values and raise them as one error? Or will I have to start the application again for every error that I'm fixing?

Reply:
ohh there is an entire section on validation. Maybe we should add a link here?
```
For more info on data validation [click here](./5-handling_deviations.md)
```

Reply:
added your suggestion